src/ponyfill.js

```diff
@@ -28,33 +28,34 @@ InstrumentedPromise.__proto__ = OriginalPromise;
 InstrumentedPromise.prototype.__proto__ = OriginalPromise.prototype;
 
 InstrumentedPromise.prototype.then = function then(onFulfilled, onRejected) {
-  return OriginalPromise.prototype.then.call(this, onFulfilled, arg => {
-    this._handled = true;
-    return onRejected(arg);
-  });
+  const next = OriginalPromise.prototype.then.call(this, onFulfilled, onRejected);
+  this._hasDownstreams = true;
+  return next;
 };
 
 function dispatchUnhandledRejectionEvent(promise, reason) {
-  const event = document.createEvent('Event');
-  /**
-   * Note: these properties should not be enumerable, which is the default setting
-   */
-  Object.defineProperties(event, {
-    promise: {
-      value: promise,
-      writable: false
-    },
-    reason: {
-      value: reason,
-      writable: false
-    }
-  });
-  event.initEvent(
-    'unhandledrejection', // Define that the event name is 'unhandledrejection'
-    false, // PromiseRejectionEvent is not bubbleable
-    true // PromiseRejectionEvent is cancelable
-  );
-  window.dispatchEvent(event);
+  if (typeof window !== 'undefined') {
```

Comment:
I think the assumption of this polyfill is that it's only meant to be used in a browser environment. Is this check just for the purposes of testing?